domain/storageprovisioning/kind.go

```diff
@@ -0,0 +1,18 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storageprovisioning
+
+// Kind represents a unique value to signal what kind a storage is within the
+// model. This value is tightly coupled to the storage kind values used for
+// charms.
+type Kind int
```

Comment:
As discussed offline, we'll merge this with domain/storage's representation in another PR.